### PR TITLE
Resolves #16 #17

### LINKED ISSUE #16: Randomize values for invalid params requests
Specific example is #shouldFailHandshakeWhenRequestHeaderUpgradeNotWebSocket should have random base64 instead of hardcoded value "not websocket".  Other cases may be present

### PR DESCRIPTION
**Pull request by dpwspoon**

### CODE REVIEW

File: src/main/java/org/kaazing/specification/ws/Functions.java [hunk 1]
```diff
@@ -90,6 +90,21 @@
         return Arrays.copyOfRange(original, from, to);
     }
 
+    @Function
+    public static String randomCase(String string) {
```
Comment (jfallows) [inline src/main/java/org/kaazing/specification/ws/Functions.java]: Suggest "text" instead of "string" as parameter name.

**Comment by jfallows** [comment]
I think we might want to restructure the pairings as follows:
```
request.headers.random.case/
  handshake.request.rpt     [writes random case, reads strict case]
  handshake.response.rpt    [reads case insensitive, writes strict case]
```
This scenario would be used by server implementations, with the handshake.response.rpt script guiding the expected behavior of the server implementation.
```
response.headers.random.case/
  handshake.request.rpt     [writes strict case, reads case insensitive]
  handshake.response.rpt    [reads strict case, writes random case]
```
This scenario would be used by client implementations, with the handshake.request.rpt script guiding the expected behavior of the client implementation.

**Comment by dpwspoon** [comment]
Resolved code review suggestions and added fix for fix for #17 with this pull request

File: src/main/java/org/kaazing/specification/ws/Functions.java [hunk 2]
```diff
@@ -90,6 +92,47 @@
...
+            } else {
+                c = Character.toLowerCase(c);
+            }
+            result.append(c);
+        }
+        return result.toString();
+    }
+
+    @Function
+    public static String randomHeaderValueOfNotCaseInsensitive(String header) {
```
Comment (jfallows) [inline src/main/java/org/kaazing/specification/ws/Functions.java]: Perhaps something shorter and more general, like `randomBase64(text)` ? :smiley:

File: src/main/java/org/kaazing/specification/ws/Functions.java [hunk 3]
```diff
@@ -90,6 +92,47 @@
...
+            int randomHeaderLength = RANDOM.nextInt(MAX_ACCEPTABLE_HEADER_LENGTH) + 1;
+            for (int i = 0; i < randomHeaderLength; i++) {
+                result.append(commonHeaderChars.charAt(RANDOM.nextInt(commonHeaderChars.length())));
+            }
+        } while (result.toString().equalsIgnoreCase(header));
+        return result.toString();
+    }
+
+    @Function
+    public static String randomHttpRequestMethodOfNot(String method) {
```
Comment (jfallows) [inline src/main/java/org/kaazing/specification/ws/Functions.java]: Perhaps rename to `randomMethodNot(method)` ?

**Pull request merged**